docs/via_guides/data-flow.md

```diff
@@ -0,0 +1,872 @@
+publication to the data availability layer (Bitcoin). As explained in its
+[README](../../core/node/via_da_dispatcher/README.md), it extends the functionality of the original zkSync da_dispatcher
+but with specific adaptations for Bitcoin:
+
+- It retrieves sealed L1 batches that are ready for dispatch from the database
+- It processes batch commitments and prepares them for publication
+- It handles both L1 batch commitments and L1 batch proofs for the Bitcoin data availability layer
+
+The main execution loop in [`da_dispatcher.rs`](../../core/node/via_da_dispatcher/src/da_dispatcher.rs) performs three
+concurrent tasks:
+
+1. `dispatch()` - Prepares and dispatches batch data to the data availability layer
+2. `poll_for_inclusion()` - Checks if dispatched data has been successfully included
+3. `dispatch_proofs()` - Handles proof dispatching for verified batches
+
+Once the batch data is prepared, the dispatcher creates entries in the database that serve as indirect tasks for the
+`via_btc_sender` component.
+
+**Implementation Details**:
+
+The DA dispatcher includes several important implementation features not covered in the high-level description:
+
+- **Robust Retry Mechanism**: The dispatcher implements a retry system for handling transient failures when dispatching
+  blobs:
+
+  ```rust
+  let dispatch_response = retry(self.config.max_retries(), batch.l1_batch_number, || {
+      self.client.dispatch_blob(batch.l1_batch_number.0, batch.pubdata.clone())
+  }).await
+  ```
+
+- **Detailed Metrics Collection**: The dispatcher tracks extensive metrics about its operations:
+
+  ```
+  METRICS.last_dispatched_l1_batch.set(batch.l1_batch_number.0 as usize);
+  METRICS.blob_size.observe(batch.pubdata.len());
+  ```
+
+- **Dual Proof Handling Paths**: The dispatcher has separate code paths for real and dummy proofs:
+  ```rust
+  async fn dispatch_proofs(&self) -> anyhow::Result<()> {
+      match self.dispatch_real_proof {
+          true => self.dispatch_real_proofs().await?,
+          false => self.dispatch_dummy_proofs().await?,
+      }
+      Ok(())
+  }
+  ```
+
+#### Bitcoin Sender
+
+The [`via_btc_sender`](../../core/node/via_btc_sender) component is responsible for inscribing the L1 batch commitment
+data onto the Bitcoin blockchain. This component:
+
+- Retrieves pending inscription requests from the database
+- Constructs Bitcoin inscription messages containing the batch commitment data
+- Creates and signs Bitcoin transactions for the inscriptions
+- Broadcasts the transactions to the Bitcoin network
+- Updates the database with the transaction information
+
+The inscription process is handled by the
+[`btc_inscription_manager.rs`](../../core/node/via_btc_sender/src/btc_inscription_manager.rs), which implements the core
+logic for creating and managing inscriptions. It works in a loop that:
+
+1. Checks for pending inscription requests in the database
```

Comment:
These are practically the same steps already explained above about the `via_btc_sender` in general